pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -40,21 +44,45 @@
 
 public class PartitionedProducerImpl extends ProducerBase {
 
+    private static final Random RANDOM = new Random(System.currentTimeMillis());
```

Comment:
if `PartitionedProducerImpl` is using it only once then should we need it as a class variable? if not then can we just keep its scope at line ln#80.?

Reply:
I was thinking about the case when you have a client producing to multiple topics, you will end up creating multiple random instances. I can switch to use ThreadLocalRandom if there is a concern using a static instance.

Reply:
@rdhabalia I changed to use `ThreadLocalRandom`. let me know if it is okay or not.